mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditor.java

```diff
@@ -201,7 +201,13 @@ void loadDocument(InputStream in, String encoding, DocumentListener documentList
     }
 
     private void write(OutputStream out) throws IOException {
-        textEditorImpl.write(new BOMWriter(out, textViewerDelegate.getEncoding()));
+        boolean isUtf8 = "UTF-8".equalsIgnoreCase(textViewerDelegate.getEncoding());
+        textEditorImpl.write(new BOMWriter(out, textViewerDelegate.getEncoding()) {
+            {
+                // Don't write BOM for utf-8 - see: https://github.com/mucommander/mucommander/issues/835
+                bomWriteChecked = isUtf8;
+            }
+        });
```

Comment:
so why not to just use OutputStreamWriter in case of UTF-8?

Reply:
fair remark :)